### PR TITLE
Add and handle billing data service

### PR DESCRIPTION
**Pull request by mcalmer**
## What does this PR change?

The billing-data-service is needed for SUMA PAYG Public Cloud image.
As we do not want a special container, we install it by default inside of the container and enable the service.

The service has a condition to only start when ISPAYG environment variable is set. This should be done by "mgradm install"
via the service file.

## GUI diff

No difference.

- [x] **DONE**

## Documentation
- No documentation needed: only internal and user invisible changes

- [x] **DONE**

## Test coverage
- No tests: already covered

- [x] **DONE**

## Links

Issue(s): https://github.com/SUSE/spacewalk/issues/24287

- [x] **DONE**

## Changelogs

Make sure the changelogs entries you are adding are compliant with https://github.com/uyuni-project/uyuni/wiki/Contributing#changelogs and https://github.com/uyuni-project/uyuni/wiki/Contributing#uyuni-projectuyuni-repository

If you don't need a changelog check, please mark this checkbox:

- [ ] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_pgsql_tests"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

# Before you merge

Check [How to branch and merge properly](https://github.com/uyuni-project/uyuni/wiki/How-to-branch-and-merge-properly)!

### CODE REVIEW

**Review by lucidd — APPROVED** [review]

**Review by rjmateus — APPROVED** [review]

File: spacewalk/admin/spacewalk-admin.changes.mc.add-and-handle-billing-data-service [hunk 1]
```diff
@@ -0,0 +1 @@
+- remove enforcing billing-adapter-service as it is running outside of the container
```
Comment (deneb-alpha) [inline spacewalk/admin/spacewalk-admin.changes.mc.add-and-handle-billing-data-service:1]:
```suggestion
- Remove enforcing billing-adapter-service as it is running outside of the container
```

File: spacewalk/admin/mgr-check-payg.service [hunk 2]
```diff
@@ -2,6 +2,7 @@
 Description=Check and install payg billing service.
```
Comment (deneb-alpha) [inline spacewalk/admin/mgr-check-payg.service:2]:
```suggestion
Description=Check and install PAYG billing service.
```

File: python/billingdataservice/billing-data-service.changes.mc.add-and-handle-billing-data-service [hunk 3]
```diff
@@ -0,0 +1,2 @@
+- make service depending on environment valiable
+- enable service by default
```
Comment (deneb-alpha) [inline python/billingdataservice/billing-data-service.changes.mc.add-and-handle-billing-data-service:2]:
```suggestion
- Make billing-data-service depending on environment variable
- Enable service by default
```

File: containers/server-image/server-image.changes.mc.add-and-handle-billing-data-service [hunk 4]
```diff
@@ -0,0 +1 @@
+- install billing-data-service in the container required for PAYG
```
Comment (deneb-alpha) [inline containers/server-image/server-image.changes.mc.add-and-handle-billing-data-service:1]:
```suggestion
- PAYG requires to install billing-data-service in the container
```

**Review by deneb-alpha — COMMENTED** [review]
I left a few comments

File: python/billingdataservice/billing-data-service.changes.mc.add-and-handle-billing-data-service [hunk 5]
```diff
@@ -0,0 +1,2 @@
+- Make billing-data-service depending on environment variable
+- Enable service by default
```
Comment (cbosdo) [inline python/billingdataservice/billing-data-service.changes.mc.add-and-handle-billing-data-service:2]: Could you please group those together to be more user-friendly? Like this:
```suggestion
- Add billing-data-service depending on environment variable
- Enable service by default
```

**Review by cbosdo — COMMENTED** [review]
This looks good to me. Do you already have a PR for uyuni-tools setting the `ISPAYG` variable?

**Comment by mcalmer** [comment]
> This looks good to me. Do you already have a PR for uyuni-tools setting the `ISPAYG` variable?

No, and I will wait with merging this one until we have it.

**Pull request merged**